pmd-core/src/main/java/net/sourceforge/pmd/cpd/FileReporter.java

```diff
@@ -40,18 +38,9 @@ public FileReporter(File reportFile, String encoding) {
 
     public void report(String content) throws ReportException {
         try {
-            Writer writer = null;
-            try {
-                OutputStream outputStream;
-                if (reportFile == null) {
-                    outputStream = System.out;
-                } else {
-                    outputStream = new FileOutputStream(reportFile);
-                }
-                writer = new BufferedWriter(new OutputStreamWriter(outputStream, encoding));
+            try (OutputStream outputStream = reportFile == null ? System.out : new FileOutputStream(reportFile);
```

Comment:
I don't think, that this way, it's easier to read.
How about leaving the OutputStream outside of the try in it's old verbose format and just have the try around the writer? if the writer is closed, then the underlying stream *should* be closed, too.

Alternatively, we could write this:

```
OutputStream outputStream;
// ... same as before

try (OutputStream out = outputStream; Writer writer = new BufferedWriter(....
```

That way, we can still figure out the outputstream on multiple lines, but still use the auto-close feature.

Reply:
The problem is that `FileOutputStream` is AutoCloseable`, so in the branch where we run `new FileOutputStream(reportFile)`, we were never actually closing the stream.

I've extracted the ternary operator to a private method, and referenced that private method in the try-with-resources statement.

Reply:
:+1: